src/swap/defi/thorchain/swapkit.ts

```diff
@@ -303,124 +299,74 @@ export function makeSwapKitPlugin(opts: EdgeCorePluginOptions): EdgeSwapPlugin {
-    //   [])[0]
-    // const fromCurrencyInfo = fromWallet.currencyInfo
-    // if (customFeeTemplate?.type === 'nativeAmount') {
-    //   customNetworkFee = inAssetGasRate
-    //   customNetworkFeeKey = customFeeTemplate.key
-    // } else if (fromCurrencyInfo.defaultSettings?.customFeeSettings != null) {
-    //   const customFeeSettings = asCustomFeeSettings(
-    //     fromCurrencyInfo.defaultSettings.customFeeSettings
-    //   )
-    //   // Only know about the key 'gasPrice'
-    //   const usesGasPrice = customFeeSettings.find(f => f === 'gasPrice')
-    //   if (usesGasPrice != null) {
-    //     customNetworkFee = inAssetGasRate
-    //     customNetworkFeeKey = 'gasPrice'
-    //   }
-    // }
-
-    // if (customNetworkFee == null || customNetworkFeeKey == null) {
-    //   throw new SwapCurrencyError(swapInfo, request)
-    // }
-
     let memoType: EdgeMemo['type'] = 'hex'
-    let memo = calldata.tcMemo ?? calldata.memo ?? ''
-
-    log.warn(memo)
+    let memo = ''
 
-    const contractAddress = tcDirect ? router : thorSwap.contract
-    let ethNativeAmount = nativeAmount
-    let publicAddress = thorAddress
+    const publicAddress = targetAddress
     let approvalData
 
-    if (EVM_CURRENCY_CODES[fromMainnetCode]) {
+    const evmTransaction = asMaybe(asEvmCleaner)(thorSwap.tx)
+    const cosmosTransaction = asMaybe(asCosmosCleaner)(thorSwap.tx)
+    if (evmTransaction != null) {
+      // EVM
       if (fromMainnetCode !== fromCurrencyCode) {
-        if (contractAddress == null)
-          throw new Error(`Missing router address for ${fromMainnetCode}`)
-        if (sourceTokenContractAddress == null)
+        if (sourceTokenContractAddress == null) {
           throw new Error(
             `Missing sourceTokenContractAddress for ${fromMainnetCode}`
           )
-        // Need to use ethers.js to craft a proper tx that calls Thorchain contract, then extract the data payload
-        // Token transactions send no ETH (or other EVM mainnet coin)
-        if (tcDirect) {
-          memo = await getEvmTokenData({
-            assetAddress: sourceTokenContractAddress,
-            amountToSwapWei: Number(nativeAmount),
-            contractAddress,
-            vaultAddress: thorAddress,
-            memo
-          })
-        } else {
-          if (contractMethod == null)
-            throw new Error('Invalid null contractMethod')
-          if (contractAddress == null)
-            throw new Error('Invalid null contractAddress')
-
-          memo = asString(transaction?.data)
         }
 
-        ethNativeAmount = '0'
-        publicAddress = contractAddress
+        memo = evmTransaction.data
 
         // Check if token approval is required and return necessary data field
         approvalData = await getEvmApprovalData({
-          contractAddress: tokenProxyMap[fromWallet.currencyInfo.pluginId],
+          contractAddress: asString(thorSwap.meta.approvalAddress),
```

Comment:
Optional: clean this value above and use it everywhere. ie

`const dexContractAddress = asString(thorSwap.meta.approvalAddress)`